File: gen_icon.js
// Generate AniSync app icons for all Android mipmap densities using Node.js canvas
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

const BASE = path.join(__dirname, 'packages/mobile/app/src/main/res');
const sizes = { xxxhdpi: 192, xxhdpi: 144, xhdpi: 96, hdpi: 72, mdpi: 48 };

for (const [name, s] of Object.entries(sizes)) {
    const canvas = createCanvas(s, s);
    const ctx = canvas.getContext('2d');
    const c = s / 2;
    const r = s / 2 - 1;

    // Background circle with gradient
    const grad = ctx.createRadialGradient(c, c, 0, c, c, r);
    grad.addColorStop(0, '#10b981');
    grad.addColorStop(1, '#0d9488');
    ctx.beginPath();
    ctx.arc(c, c, r, 0, Math.PI * 2);
    ctx.fillStyle = grad;
    ctx.fill();

    // Play triangle
    const ts = s * 0.32;
    const tx = c - ts * 0.2;
    const ty = c - ts / 2;
    ctx.beginPath();
    ctx.moveTo(tx, ty);
    ctx.lineTo(tx + ts, c);
    ctx.lineTo(tx, ty + ts);
    ctx.closePath();
    ctx.fillStyle = 'rgba(255,255,255,0.95)';
    ctx.fill();

    // Sync arc top
    ctx.beginPath();
    ctx.arc(c, c, s * 0.42, -Math.PI * 0.85, -Math.PI * 0.15);
    ctx.strokeStyle = 'rgba(255,255,255,0.6)';
    ctx.lineWidth = Math.max(2, s / 22);
    ctx.stroke();

    // Sync arc bottom
    ctx.beginPath();
    ctx.arc(c, c, s * 0.42, Math.PI * 0.15, Math.PI * 0.85);
    ctx.stroke();

    // Arrow tips
    const ar = s * 0.42;
    const aw = Math.max(3, s / 16);

    // Top-right arrow
    const a1 = -Math.PI * 0.15;
    const ax1 = c + ar * Math.cos(a1);
    const ay1 = c + ar * Math.sin(a1);
    ctx.beginPath();
    ctx.moveTo(ax1, ay1);
    ctx.lineTo(ax1 - aw * 0.8, ay1 - aw);
    ctx.lineTo(ax1 + aw * 0.3, ay1 - aw * 0.5);
    ctx.closePath();
    ctx.fillStyle = 'rgba(255,255,255,0.6)';
    ctx.fill();

    // Bottom-left arrow
    const a2 = Math.PI * 0.85;
    const ax2 = c + ar * Math.cos(a2);
    const ay2 = c + ar * Math.sin(a2);
    ctx.beginPath();
    ctx.moveTo(ax2, ay2);
    ctx.lineTo(ax2 + aw * 0.8, ay2 + aw);
    ctx.lineTo(ax2 - aw * 0.3, ay2 + aw * 0.5);
    ctx.closePath();
    ctx.fill();

    const outDir = path.join(BASE, `mipmap-${name}`);
    fs.mkdirSync(outDir, { recursive: true });
    const buf = canvas.toBuffer('image/png');
    fs.writeFileSync(path.join(outDir, 'ic_launcher.png'), buf);
    console.log(`  ${name}: ${s}x${s}`);
}
console.log('Icons generated!');

File: gen_icon.py
from PIL import Image, ImageDraw
import math, os

BASE = "C:/Users/emin/.gemini/antigravity/scratch/anisync/packages/mobile/app/src/main/res"
sizes = {"xxxhdpi": 192, "xxhdpi": 144, "xhdpi": 96, "hdpi": 72, "mdpi": 48}

for name, s in sizes.items():
    img = Image.new("RGBA", (s, s), (0, 0, 0, 0))
    d = ImageDraw.Draw(img)
    c = s // 2
    r = s // 2 - 1

    # Background - emerald/cyan gradient circle
    for i in range(r, 0, -1):
        ratio = i / r
        cr = int(10 + (6 - 10) * (1 - ratio))
        cg = int(185 + (182 - 185) * (1 - ratio))
        cb = int(129 + (212 - 129) * (1 - ratio))
        d.ellipse([c - i, c - i, c + i, c + i], fill=(cr, cg, cb, 255))

    # Play triangle (centered, white)
    ts = int(s * 0.32)
    tx = c - int(ts * 0.25)
    ty = c - ts // 2
    pts = [(tx, ty), (tx + ts, c), (tx, ty + ts)]
    d.polygon(pts, fill=(255, 255, 255, 240))

    # Sync arc around
    lw = max(2, s // 20)
    ar = int(s * 0.43)
    d.arc([c - ar, c - ar, c + ar, c + ar], -40, 220, fill=(255, 255, 255, 160), width=lw)

    # Small arrow tip on arc end
    angle_rad = math.radians(-40)
    ax = c + int(ar * math.cos(angle_rad))
    ay = c + int(ar * math.sin(angle_rad))
    aw = max(3, s // 14)
    d.polygon([
        (ax, ay),
        (ax - aw, ay + aw // 2),
        (ax - aw // 2, ay - aw),
    ], fill=(255, 255, 255, 160))

    outdir = os.path.join(BASE, f"mipmap-{name}")
    os.makedirs(outdir, exist_ok=True)
    img.save(os.path.join(outdir, "ic_launcher.png"))
    print(f"  {name}: {s}x{s} -> {outdir}/ic_launcher.png")

print("Done!")
